### PR TITLE
Revert tx history to subscan

### PR DESCRIPTION
**Pull request by ERussel**
- return subscan logic for transaction history
- make single request for all rewards from subquery to calculate total rewards

### CODE REVIEW

File: fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift [hunk 1]
```diff
@@ -5,6 +5,150 @@ import IrohaCrypto
...
+
+        status = .commited
+
+        let amount: Decimal = {
+            guard let amountValue = BigUInt(item.amount) else {
+                return 0.0
+            }
+
+            return Decimal.fromSubstrateAmount(amountValue, precision: networkType.precision) ?? 0.0
+        }()
```
Comment (stas-litvinov) [inline fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift:83]: I suggest to replace it with a shorter form: 
```
let amount = Decimal.fromSubstrateAmount(
    BigUInt(extrinsic.fee) ?? 0,
    precision: asset.precision
) ?? .zero
```

File: fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift [hunk 2]
```diff
@@ -5,6 +5,150 @@ import IrohaCrypto
...
+        asset: WalletAsset,
+        addressFactory: SS58AddressFactoryProtocol
+    ) -> AssetTransactionData {
+        let amount: Decimal = {
+            guard let amountValue = BigUInt(item.fee) else {
+                return 0.0
+            }
+
+            return Decimal.fromSubstrateAmount(amountValue, precision: networkType.precision) ?? 0.0
+        }()
```
Comment (stas-litvinov) [inline fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift:118]: The same as above

File: fearless/Common/Network/Subscan/SubscanHistoryOperationFactory.swift [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            return WalletRemoteHistoryData(
+                historyItems: finalItems,
+                context: nextContext
+            )
+        }
+    }
+}
+
+extension SubscanHistoryOperationFactory: WalletRemoteHistoryFactoryProtocol {
+    func createOperationWrapper(for context: TransactionHistoryContext, address: String, count _: Int)
```
Comment (stas-litvinov) [inline fearless/Common/Network/Subscan/SubscanHistoryOperationFactory.swift:205]: `count` parameter is not used

File: fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift [hunk 4]
```diff
@@ -5,6 +5,150 @@ import IrohaCrypto
...
+            context: nil
+        )
+    }
+
+    static func createTransaction(
+        from item: SubscanRewardItemData,
+        address: String,
+        networkType: SNAddressType,
+        asset: WalletAsset,
+        addressFactory _: SS58AddressFactoryProtocol
```
Comment (stas-litvinov) [inline fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift:71]: `addressFactory` parameter is not used

**Review by stas-litvinov — APPROVED** [review]

File: fearless/Common/Network/Subscan/SubscanHistoryOperationFactory.swift (hunk 3 above)
Comment (ERussel) [reply fearless/Common/Network/Subscan/SubscanHistoryOperationFactory.swift:205]: Its part of the protocol and is used by another implementation

File: fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift (hunk 4 above)
Comment (ERussel) [reply fearless/Common/Extension/Wallet/AssetTransactionData+HistoryItemData.swift:71]: it is part of the protocol and is used by a separate implementation

**Review by stas-litvinov — APPROVED** [review]

**Pull request merged**